File: java/3list/src/com/stanislaw/quadrangles/Diamond.java
package com.stanislaw.quadrangles;
/*
@author: Stanisław Woźniak
@version: 1.0
*/
public class Diamond extends Quadrangle {

    public Diamond(double side1, double angle) {
        super(side1, side1, side1, side1, angle);
    }

    @Override
    public double field() {
        return this.side1*this.side1*Math.sin(Math.toRadians(this.angle));
    }

    @Override
    public double perimeter() {
        return 4*this.side1;
    }
}

File: C++/3list/Diamond.cpp
/*
@author: Stanisław Woźniak
@title: diamond.cpp
*/

#include "Diamond.hpp"
#include <cmath>

Diamond::Diamond(double side, double angle){
  side1 = side;
  side2 = side;
  side3 = side;
  side4 = side;
  this -> angle = angle;
}

double Diamond::field(){
  return sin((angle*M_PI)/180)*side1*side1;
}

double Diamond::perimeter(){
  return 4*side1;
}

